OctaneVSPlugin/ViewModel/SearchItemsViewModel.cs

```diff
@@ -0,0 +1,134 @@
+﻿/*!
+* (c) 2016-2018 EntIT Software LLC, a Micro Focus company
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using MicroFocus.Adm.Octane.VisualStudio.Common;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.Windows.Input;
+using Task = System.Threading.Tasks.Task;
+
+namespace MicroFocus.Adm.Octane.VisualStudio.ViewModel
+{
+    public class SearchItemsViewModel : INotifyPropertyChanged
+    {
+        private readonly OctaneServices _octaneService;
+        private string _searchFilter = string.Empty;
+        private readonly IList<BaseItemViewModel> _searchResults = new List<BaseItemViewModel>();
+
+        public SearchItemsViewModel(string searchFilter)
+        {
+            _searchFilter = searchFilter;
+
+            RefreshCommand = new DelegatedCommand(Refresh);
+
+            Mode = WindowMode.Loading;
+
+            _octaneService = new OctaneServices(
+                OctaneConfiguration.Url,
+                OctaneConfiguration.SharedSpaceId,
+                OctaneConfiguration.WorkSpaceId,
+                OctaneConfiguration.Username,
+                OctaneConfiguration.Password);
+        }
+
+        /// <summary>
+        /// Search for all entities satisfying the criteria
+        /// </summary>
+        public async Task Search()
+        {
+            try
+            {
+                _searchResults.Clear();
+
+                if (string.IsNullOrEmpty(_searchFilter))
+                {
+                    Mode = WindowMode.Loaded;
+                    return;
+                }
+
+                await _octaneService.Connect();
+                var results = await _octaneService.SearchEntities(_searchFilter, 5);
```

Comment:
Should the limit per type be 5? I'm not sure what it is on other plugins. Maybe we should align them?

Reply:
@atothhpe Increased to 20 as we discussed